File: src/main/kotlin/factory_method/java_case/idcard/IDCardFactory.java
package factory_method.java_case.idcard;

import factory_method.java_case.framwork.Factory;
import factory_method.java_case.framwork.Product;

public class IDCardFactory extends Factory {
    @Override
    protected Product createProduct(String owner) {
        return new IDCard(owner);
    }

    @Override
    protected void registerProduct(Product product) {
        System.out.println(product + "을 등록했습니다.");
    }
}

File: src/main/kotlin/factory_method/kotlin_case/idcard/IDCardFactory.kt
package factory_method.kotlin_case.idcard

import factory_method.kotlin_case.framework.Factory
import factory_method.kotlin_case.framework.Product

class IDCardFactory:Factory() {
    override fun createProduct(owner: String): Product {
        return IDCard(owner)
    }

    override fun registerProduct(product: Product) {
        println("${product}을 등록했습니다.")
    }
}